lib/spack/llnl/util/cpu/microarchitectures.json

```diff
@@ -694,6 +694,39 @@
         }
       }
     },
+    "k10": {
+      "from": "x86_64",
+      "vendor": "AuthenticAMD",
+      "features": [
+        "mmx",
+        "sse",
+        "sse2",
+        "sse4a",
+        "abm",
+        "cx16",
```

Comment:
Can we guarantee that cx16 and popcnt instructions are present on k10/barcelona? I see that gcc doesn't check for them when detecting barcelona.

Reply:
According to the [wikipedia](https://en.wikipedia.org/wiki/SSE4#POPCNT_and_LZCNT) entry, AMD added `popcnt` along with `sse4a`.

Reply:
Re `cx16` according to [what I can piece together](https://superuser.com/questions/187254/how-prevalent-are-old-x64-processors-lacking-the-cmpxchg16b-instruction?rq=1), `CMPXCHG16B` was introduced along with DDR2 support to the Athlon 64 line with Orleans, which uses the K8 microarchitecture. Everything subsequent got it because it was a [requirement of Windows 8.1 (64bit)](https://www.pcworld.com/article/2058683/new-windows-8-1-requirements-strand-some-users-on-windows-8.html).

Reply:
popcnt is included in `abm`, which we already check for. I guess at that point I don't really care whether we check for it or not when we already check for `abm`. My inclination is this is just another way that a kernel could mess us up by not exposing it.

Reply:
Since we don't check for popcnt for the other AMD chips, can we remove it here? It is definitely a part of abm.

For `cx16` I'm fine leaving it in. We check for it on all of the other chips and it looks from what you found and my own searching it should be available on `k10`.

I'm ready to approve this once you take popcnt out.

Reply:
`popcnt` has been removed.